Library/Homebrew/brew.sh

```diff
@@ -114,6 +114,25 @@ else
   : "${HOMEBREW_OS_VERSION:=$(uname -r)}"
   HOMEBREW_OS_USER_AGENT_VERSION="$HOMEBREW_OS_VERSION"
 
+  # Ensure the system Curl is a version that supports modern HTTPS certificates.
+  HOMEBREW_SYSTEM_CURL_VERSION="$($(command -v curl) --version 2>/dev/null | head -n1 | awk '{print $2}')"
```

Comment:
Would be preferable to use Bash string operators (like the line below) rather than `awk` here (and for Git below): https://stackoverflow.com/a/3162500

Reply:
Like one of these?
```
HOMEBREW_SYSTEM_CURL_VERSION="$($(command -v curl) --version 2>/dev/null | head -n1 | cut -d ' ' -f 2)"
HOMEBREW_SYSTEM_CURL_VERSION="$($(command -v curl) --version 2>/dev/null | head -c 12 | { read x; echo ${x#curl }; })"
```

Reply:
```bash
$ git_version_output=$(git --version)
$ git_version=${git_version##* }
$ echo $git_version
2.19.1
```

```bash
$ curl_version_output=$(curl --version)
$ curl_name_and_version=${curl_version_output%% \(*}
$ curl_version=${curl_name_and_version##* }
$ echo $curl_version
7.54.0
```

Reply:
(Could use this format for `HOMEBREW_CURL_VERSION` too)

Reply:
at https://github.com/Homebrew/brew/blob/d9918109f42cbca688d6432eb56603b92c9c9dc1/Library/Homebrew/brew.sh#L181

Reply:
In fact they could share the `curl` invocation, too.

Reply:
Not really, since the first `curl` in the `PATH` might differ from the one `HOMEBREW_CURL` is set to.